src/TraceEvent/TraceLoggingEventId.cs

```diff
@@ -10,19 +11,20 @@ namespace Microsoft.Diagnostics.Tracing
     /// <summary>
     /// TraceLoggingEvnetId is a class that manages assigning event IDs (small 64k numbers)
     /// to TraceLogging Style events (which don't have them).  Because TraceEvent uses EventIDs
-    /// so fundamentally this difficiency is very problematic.   
+    /// so fundamentally this deficiency is very problematic.   
     /// 
     /// Arguably this should have been done by the ETW system itself.  
     /// 
     /// You use it by calling TestForTraceLoggingEventAndFixupIfNeeded on eventRecords.  
     /// You also have to explicitly call 'Dispose' when you are done with this class.  
     /// </summary>
-    internal unsafe struct TraceLoggingEventId
+    internal unsafe struct TraceLoggingEventId : IDisposable
     {
         /// <summary>
         /// Checks to see if eventRecord has TraceLogging meta data associated with it (EVENT_HEADER_EXT_TYPE_EVENT_SCHEMA_TL)
         /// and if so updates EventHeader.Id to be an event ID unique to that provider/opcode/meta-data blob. 
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
@northtyphoon Currently the library does not support .NET 3.5, and there is no concrete plan to do so. My suggestion here is to only include this change if it is needed, e.g. if:

1. PerfView revealed this method in traces (a clear indication that not only is it not being inlined, but it's heavily used)
2. You saw this method in disassembled JIT code and observed that it wasn't inlined, and knew separately that it's heavily used so it should be
3. Some other indicator suggesting that the attribute provides measurable value

For me, the bar here is low. I just don't want to include it if no one will notice that it was included, because it makes things a tiny bit more complicated if we ever decide to go back and address the specific request I linked to.